codegen/src/test/resources/software/amazon/awssdk/codegen/poet/model/serviceclientconfiguration-builder.java

```diff
@@ -0,0 +1,182 @@
+package software.amazon.awssdk.services.jsonprotocoltests.internal;
+
+import java.net.URI;
+import software.amazon.awssdk.annotations.Generated;
+import software.amazon.awssdk.annotations.SdkInternalApi;
+import software.amazon.awssdk.awscore.client.config.AwsClientOption;
+import software.amazon.awssdk.core.client.config.ClientOverrideConfiguration;
+import software.amazon.awssdk.core.client.config.SdkClientConfiguration;
+import software.amazon.awssdk.core.client.config.SdkClientOption;
+import software.amazon.awssdk.core.client.config.internal.SdkClientConfigurationUtil;
+import software.amazon.awssdk.endpoints.EndpointProvider;
+import software.amazon.awssdk.http.auth.spi.scheme.AuthSchemeProvider;
+import software.amazon.awssdk.identity.spi.AwsCredentialsIdentity;
+import software.amazon.awssdk.identity.spi.IdentityProvider;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.jsonprotocoltests.JsonProtocolTestsServiceClientConfiguration;
+import software.amazon.awssdk.services.jsonprotocoltests.auth.scheme.JsonProtocolTestsAuthSchemeProvider;
+import software.amazon.awssdk.utils.Validate;
+
+@Generated("software.amazon.awssdk:codegen")
+@SdkInternalApi
+public class JsonProtocolTestsServiceClientConfigurationBuilder {
+    public static JsonProtocolTestsServiceClientConfiguration.Builder builder() {
+        return new BuilderImpl();
+    }
+
+    public static BuilderInternal builder(SdkClientConfiguration.Builder builder) {
+        return new BuilderImpl(builder);
+    }
+
+    public interface BuilderInternal extends JsonProtocolTestsServiceClientConfiguration.Builder {
+        SdkClientConfiguration buildSdkClientConfiguration();
+    }
+
+    public static class BuilderImpl implements BuilderInternal {
+        private final SdkClientConfiguration.Builder builder;
```

Comment:
Can we call this builder something that distinguishes it from this type? configBuilder, internalBuilder, delegatingBuilder?

Reply:
I meant the variable `builder` that represents the SdkClientConfiguration.Builder

Reply:
Sure, just tell me how to name it and I will make the change.

Reply:
Go with `internalBuilder` then?